exchange/src/exchange/rabbitmq/order_handler/RabbitMQOrderHandler.cpp

```diff
@@ -10,7 +10,7 @@ namespace rabbitmq {
 void
 RabbitMQOrderHandler::handle_incoming_market_order(
     engine_manager::EngineManager& engine_manager, manager::ClientManager& clients,
-    MarketOrder&& order
+    MarketOrder&& order, bool is_rmq
```

Comment:
Pass the engine state not just a bool

Also you could make this a template arg and let the compiler optimize it out.